app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.test.tsx

```diff
@@ -66,36 +66,27 @@ const Subject = ({
 
 describe("AppLayoutSidebarLocked", () => {
   describe("navigation links", () => {
-    it("when live, hides edit and review links, displays design link and disabled results item", () => {
+    it("when live, hides edit and review links, displays summary link and disabled results item", () => {
       render(<Subject status={NimbusExperimentStatus.LIVE} />);
       [
-        "edit-overview",
-        "edit-branches",
-        "edit-metrics",
-        "edit-audience",
-        "edit-request-review",
-      ].forEach((slug) => {
-        expect(screen.queryByTestId(`nav-${slug}`)).not.toBeInTheDocument();
+        "Overview",
+        "Branches",
+        "Metrics",
+        "Audience",
+        "Review & Launch",
+      ].forEach((text) => {
+        expect(
+          screen.queryByText(text, { selector: ".navbar a" }),
```

Comment:
I'm glad you brought this up because this is an approach I was going to start leaning into in upcoming tests/refactors, so good to chat about it.

> I think the general react-testing-library advice is to prefer exercising the UI like a human would

You nailed it - I am hoping to start abiding by the RTL [query priority](https://testing-library.com/docs/queries/about#priority) list, where the elements we're looking up are accessible to the user, and saving testids for things like checking if whole components are rendered. Fragile, I can see that, but I suppose it also makes it a more confident test no? An alternative to looking up by text could be to look up by role, if you'd prefer?

Thanks for raising this!

Reply:
Sounds like an ADR folks!

Reply:
I don't know that I have a strong opinion either way here, but can we consider a selectors constants object if this will be a pattern we'll be using? That way if tests do break due to an HTML change, there's just one spot to update.

Reply:
> can we consider a selectors constants object if this will be a pattern we'll be using?

Yeah, this might be what you're already referencing, but the content-server for FxA does this for integration tests:

https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/tests/functional/lib/selectors.js

Might be handy for our unit tests if we switch to text and selectors. I'm still kind of skeptical on what the additional complexity gets us vs testids, but not enough to actually object to trying it

Reply:
> Sounds like an ADR folks!

I don't know that we necessarily need an ADR.

To be fair, I'd say Jody's changes more closely conform to documented recommendations for react-testing-library - and we essentially have [a previous architectural decision](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0002-nimbus-front-end-directory-tree-and-stack.md) to use that.

If we kept doing what I prefer, that might be (or should have been 😅) worth an ADR